python/python/lance/dataset.py

```diff
@@ -1730,6 +1756,50 @@ def create_index(
                 )
             kwargs["num_sub_vectors"] = num_sub_vectors
 
+            if (
+                pq_codebook is None
+                and accelerator is not None
+                and "precomputed_partitions_file" in kwargs
+            ):
+                partitions_file = kwargs["precomputed_partitions_file"]
+                del kwargs["precomputed_partitions_file"]
+                import lance
+
+                partitions_ds = lance.dataset(partitions_file)
+                # Use accelerator to train pq codebook
+                from .vector import (
+                    compute_pq_codes,
+                    train_pq_codebook_on_accelerator,
+                )
```

Comment:
these don't need lazy import right?

Reply:
this isn't fixed yet

Reply:
I think I mixed this up with a different lazy import sorry.
In this case, it turns out they do - it resolves a circular import. We do a similar thing for ivf assignments.